src/rt-app_taskgroups.c

```diff
@@ -0,0 +1,381 @@
+#include <stdlib.h>
+#include <string.h>
+#include <mntent.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <stdio.h>
+#include <errno.h>
+
+#include "config.h"
+#include "rt-app_utils.h"
+#include "rt-app_taskgroups.h"
+
+#define PIN "[tg] "
+
+typedef struct _taskgroup_ctrl_t
+{
+	char *mount_point;
+	taskgroup_data_t *tg_array;
+	unsigned int nr_tgs;
+} taskgroup_ctrl_t;
+
+static taskgroup_ctrl_t ctrl;
+
+const static unsigned int max_nbr_tgs = 32;
```

Comment:
This limitation should be written in the tutorial

Reply:
I added:

```diff --git a/doc/tutorial.txt b/doc/tutorial.txt
index 8650cbfa7105..8d0e2e458ed5 100644
--- a/doc/tutorial.txt
+++ b/doc/tutorial.txt
@@ -266,6 +266,11 @@ policy other than SCHED_OTHER and without taskgroup data can coexist in the
 setup. An empty taskgroup ("") is allowed and is interpreted as if there is
 no taskgroup data given. Pre-existing parts of a taskgroup are preserved.
 They are not removed during rt-app teardown.
+The default build supports up to 32 different taskgroups. In case a
+configuration exceeds this limit the error message '[rt-app] <error> [tg] #
+taskgroups exceeds max # taskgroups [32]' is issued before rt-app exits. The
+number of supported taskgroups can be increased by changing 'const static
+unsigned int max_nbr_tgs = 32' at compile time.
 
 *** events ***```